lightningd/options.c

```diff
@@ -1117,6 +1137,9 @@ static void register_opts(struct lightningd *ld)
 	opt_register_arg("--announce-addr", opt_add_announce_addr, NULL,
 			 ld,
 			 "Set an IP address (v4 or v6) or .onion v3 to announce, but not listen on");
+	opt_register_noarg("--disable-ip-discovery", opt_set_invbool,
+			 &ld->config.announce_discovered_ip,
+			 "Turn off address announcement of detected public IP");
```

Comment:
Just a note: I usually prefer positive options, not negative, with the default being `!ld->always_use_proxy`.

In theory always-use-proxy could be a normal SOCKS proxy, and for all we know you've set up a reverse tunnel.  But that's a really weird case (and you can just advertize an explicit IP for that).

Reply:
Hm, if I make it a positive bool option, then ip announcement can't be enabled by default, otherwise the switch would do nothing. Or do I get somthing wrong? What I could do is rename the internal code variables also to `disable_ip_discovery` which defaults to `false` and use an `opt_set_bool` instead of `opt_set_invbool`. Do you meant that?

Reply:
I renamed the internal variable also to `disable_ip_discovery` defaulting to `false`. This is now a switch, not an bool option. The switch effectively turns of ip discovery anouncements by using `opt_set_bool` instead of `opt_set_invbool`. 

Can you re-check?